### PR TITLE
[WIP]Add antrea-interworking package

### PR DESCRIPTION
**Pull request by wenqiq**
Signed-off-by: Wenqi Qiu <wenqiq@vmware.com>

### What this PR does / why we need it
This pr extends antrea package with interworking, when antreaNsx is enabled in antreaConfig, interworking will also be reconciled by kapp-controller
### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### CODE REVIEW

File: addons/controllers/antrea/antreaconfig_controller.go [hunk 1]
```diff
@@ -91,6 +91,13 @@ func (r *AntreaConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return retResult, err
 	}
 
+	if antreaConfig.Spec.Antrea.AntreaConfigDataValue.AntreaNsx.Enable {
```
Comment (liu4480) [inline addons/controllers/antrea/antreaconfig_controller.go:94]: I think this reconciler is not needed

File: apis/addonconfigs/cni/v1alpha1/antrea-interworking_config_types.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type AntreaInterworkingON struct {
```
Comment (liu4480) [inline apis/addonconfigs/cni/v1alpha1/antrea-interworking_config_types.go:7]: this crd is not needed, you only need to define the package, other than defining a new crd

File: addons/controllers/antrea/antrea-interworking_util.go [hunk 3]
```diff
@@ -0,0 +1,207 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package controllers
+
+import (
```
Comment (liu4480) [inline addons/controllers/antrea/antrea-interworking_util.go:6]: I think this file is not needed

File: apis/addonconfigs/config/crd/bases/cni.tanzu.vmware.com_antrea-interworkingconfigs.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.9.2
+  creationTimestamp: null
+  name: antrea-interworkingconfigs.cni.tanzu.vmware.com
```
Comment (liu4480) [inline apis/addonconfigs/config/crd/bases/cni.tanzu.vmware.com_antrea-interworkingconfigs.yaml:8]: I think this crd is not needed

File: addons/controllers/antrea/antreaconfig_util.go [hunk 5]
```diff
@@ -227,5 +267,8 @@ func mapAntreaConfigSpec(cluster *clusterv1beta1.Cluster, config *cniv1alpha1.An
 	configSpec.Antrea.AntreaConfigDataValue.FeatureGates.MultiCluster = config.Spec.Antrea.AntreaConfigDataValue.FeatureGates.MultiCluster
 	configSpec.Antrea.AntreaConfigDataValue.FeatureGates.SecondaryNetwork = config.Spec.Antrea.AntreaConfigDataValue.FeatureGates.SecondaryNetwork
 	configSpec.Antrea.AntreaConfigDataValue.FeatureGates.TrafficControl = config.Spec.Antrea.AntreaConfigDataValue.FeatureGates.TrafficControl
+
+	//todo add nsx config here once antreaNsx ia packaged into antrea
```
Comment (liu4480) [inline addons/controllers/antrea/antreaconfig_util.go:271]: you should also remove antrea Nsx pr and rebase to latest main as antrea Nsx pr is merged into main, I should also add nsx config here

**Comment by wenqiq** [comment]
Close this PR as the base branch need to change to a normal repo branch, and move to #4510

**Pull request closed without merging**